packages/compass-import-export/src/modules/export.ts

```diff
@@ -1,676 +1,237 @@
- */
-export const updateSelectedFields = (fields: ExportFieldsType) => ({
-  type: UPDATE_SELECTED_FIELDS,
-  fields: fields,
-});
-
-/**
- * Update export fields (list of full field names)
- * @api public
- * @param {Object} fields: currently selected/disselected fields to be exported
- */
-export const updateAllFields = (fields: ExportFieldsType) => ({
-  type: UPDATE_ALL_FIELDS,
-  fields: fields,
-});
-
-/**
- * Select fields to be exported
- * @api public
- * @param {String} status: next step in export
- */
-export const changeExportStep = (step: ExportStep) => ({
-  type: CHANGE_EXPORT_STEP,
-  status: step,
-});
-
-const fetchDocumentCount = async (
-  dataService: DataService,
-  ns: string,
-  query: ExportQueryType
-) => {
-  // When there is no filter/limit/skip try to use the estimated count.
-  if (
-    (!query.filter || Object.keys(query.filter).length < 1) &&
-    !query.limit &&
-    !query.skip
-  ) {
-    try {
-      const count = await dataService.estimatedCount(ns, {});
-      return count;
-    } catch (estimatedCountErr) {
-      // `estimatedDocumentCount` is currently unsupported for
-      // views and time-series collections, so we can fallback to a full
-      // count in these cases and ignore this error.
-    }
-  }
-
-  const count = dataService.count(ns, query.filter || {}, {
-    ...(query.limit ? { limit: query.limit } : {}),
-    ...(query.skip ? { skip: query.skip } : {}),
-  });
-  return count;
-};
-
-/**
- * Open the export modal.
- * Counts the documents to be exported given the current query on modal open to
- * provide user with accurate export data
- *
- * @api public
- */
-export const openExport =
-  ({
-    namespace,
-    query,
-    count: maybeCount,
-    aggregation,
-  }: {
-    namespace: string;
-    query: ExportQueryType;
-    // Optional pre supplied count to shortcut and
-    // avoid a possibly expensive re-count.
-    count?: number;
-    aggregation?: ExportAggregationType;
-  }): ThunkAction<Promise<void>, RootExportState, void, AnyAction> =>
-  async (dispatch, getState) => {
-    const isAggregation = !!aggregation;
-    track('Export Opened', { type: isAggregation ? 'aggregation' : 'query' });
-    const {
-      dataService: { dataService },
-    } = getState();
-
-    let count = null;
-    try {
-      count =
-        maybeCount ??
-        (!isAggregation
-          ? await fetchDocumentCount(dataService!, namespace, query)
-          : null);
-    } catch (e: unknown) {
-      dispatch(onError(e as Error));
-    }
-    dispatch(nsChanged(namespace));
-    dispatch(onModalOpen({ namespace, query, count, aggregation }));
+import type { SchemaPath } from '../export/gather-fields';
+import type {
+  ExportAggregation,
+  ExportQuery,
+  ExportResult,
+} from '../export/export-types';
+import { queryHasProjection } from '../utils/query-has-projection';
+
+// TODO(COMPASS-6426): Update the fields to export type based on the gather fields result type.
+export type FieldsToExport = {
```

Comment:
As noted in the description of this pr, once the gatherFields pr lands I'll update how we're selecting fields. Currently changing selected fields is broken here. And we're using a fairly hacky symbol separator here for fields that I'd like to remove.